File: onyx-database-tests/src/entities/partition/ToManyPartitionEntityChild.java
package entities.partition;

import com.onyx.persistence.IManagedEntity;
import com.onyx.persistence.ManagedEntity;
import com.onyx.persistence.annotations.*;

import java.util.List;

/**
 * Created by timothy.osborn on 3/5/15.
 */
@Entity(fileName = "web/partition")
public class ToManyPartitionEntityChild extends ManagedEntity implements IManagedEntity {

    @Identifier(generator = IdentifierGenerator.SEQUENCE)
    @Attribute
    public Long id;

    @Attribute
    @Partition
    public Long partitionId;

    @Relationship(type = RelationshipType.MANY_TO_MANY, inverse = "child", inverseClass = ToManyPartitionEntityParent.class, cascadePolicy = CascadePolicy.ALL, fetchPolicy = FetchPolicy.LAZY)
    public List<ToManyPartitionEntityParent> parent;

}

File: onyx-database-tests/src/main/java/entities/partition/ToManyPartitionEntityChild.kt
package entities.partition

import com.onyx.persistence.IManagedEntity
import com.onyx.persistence.ManagedEntity
import com.onyx.persistence.annotations.*
import com.onyx.persistence.annotations.values.CascadePolicy
import com.onyx.persistence.annotations.values.FetchPolicy
import com.onyx.persistence.annotations.values.IdentifierGenerator
import com.onyx.persistence.annotations.values.RelationshipType

/**
 * Created by timothy.osborn on 3/5/15.
 */
@Entity(fileName = "web/partition")
class ToManyPartitionEntityChild : ManagedEntity(), IManagedEntity {

    @Identifier(generator = IdentifierGenerator.SEQUENCE)
    @Attribute
    var id: Long? = null

    @Attribute
    @Partition
    var partitionId: Long? = null

    @Relationship(type = RelationshipType.MANY_TO_MANY, inverse = "child", inverseClass = ToManyPartitionEntityParent::class, cascadePolicy = CascadePolicy.ALL, fetchPolicy = FetchPolicy.LAZY)
    var parent: MutableList<ToManyPartitionEntityParent>? = null

}
